kedro/io/data_catalog.py

```diff
@@ -615,15 +616,50 @@ def add_transformer(
                 )
             self._transformers[data_set_name].append(transformer)
 
-    def list(self) -> List[str]:
-        """List of ``DataSet`` names registered in the catalog.
+    def list(self, regex_search: Optional[str] = None) -> List[str]:
+        """
+        List of ``DataSet`` names registered in the catalog if called with
+        default arguments, this can be filtered by providing an optional regular
+        expression which will only return matching keys.
```

Comment:
```suggestion
        List of all ``DataSet`` names registered in the catalog.
        This can be filtered by providing an optional regular expression 
        which will only return matching keys.
```